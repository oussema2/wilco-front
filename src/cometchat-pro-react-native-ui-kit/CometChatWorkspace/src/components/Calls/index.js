export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
