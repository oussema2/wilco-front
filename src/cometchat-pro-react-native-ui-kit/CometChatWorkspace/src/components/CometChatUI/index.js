/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
