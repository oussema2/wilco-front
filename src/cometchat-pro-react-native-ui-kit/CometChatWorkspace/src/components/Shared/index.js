export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
