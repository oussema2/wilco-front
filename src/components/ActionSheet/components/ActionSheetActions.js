import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
