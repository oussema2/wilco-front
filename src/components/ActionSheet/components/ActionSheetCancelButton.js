import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
