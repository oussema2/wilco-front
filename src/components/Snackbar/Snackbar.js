import React from 'react';
import { observer } from 'mobx-react';
import { Snackbar as SnackbarComponent } from 'react-native-paper';
import { useSnackbarService } from '../../providers/SnackbarProvider';
import { defaultStyles, successStyles, errorStyles } from './Snackbar.styles';

const Snackbar = ( ) => {
	const snackbarService = useSnackbarService();

	const onDismissSnackBar = () => snackbarService.hide();

	let styles = defaultStyles;
	if ( snackbarService.type === 'success' ) { styles = { ...styles, ...successStyles }; }
	if ( snackbarService.type === 'error' ) { styles = { ...styles, ...errorStyles }; }

	return (
		<SnackbarComponent
			style={styles.snackbar}
			visible={snackbarService.isVisible}
			onDismiss={onDismissSnackBar}
			duration={3000}
		>
			{snackbarService.message}
		</SnackbarComponent>
	);
};

Snackbar.propTypes = {
};

Snackbar.defaultProps = {
};

export default observer( Snackbar );
