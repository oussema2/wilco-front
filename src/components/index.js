export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
