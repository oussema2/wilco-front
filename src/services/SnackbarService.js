import { makeAutoObservable } from 'mobx';

export default class SnackbarService {
	constructor() {
		this.isVisible = false;
		this.message = '';

		makeAutoObservable( this );
	}

	_show( config ) {
		this.isVisible = true;
		this.message = config.message;
	}

	showInfo( config ) {
		this.type = 'info';
		this._show( config );
	}

	showSuccess( config ) {
		this.type = 'success';
		this._show( config );
	}

	showError( config ) {
		this.type = 'error';
		this._show( config );
	}

	hide() {
		this.isVisible = false;
	}
}
