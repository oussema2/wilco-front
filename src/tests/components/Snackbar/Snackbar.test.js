import React from 'react';
import { act, render } from '@testing-library/react-native';
import { Snackbar } from '../../../components/Snackbar';
import mockUseSnackbarService from '../../mocks/mockUseSnackbarService';
import SnackbarService from '../../../services/SnackbarService';

describe( '<Snackbar />', () => {
	let component;
	const snackbarService = new SnackbarService();
	const message = 'A message';
	const config = { message };

	const setUp = () => {
		component = render( <Snackbar /> );
	};

	const showSnackbar = () => {
		act( () => {
			snackbarService.showInfo( config );
		} );
	};

	beforeEach( () => {
		jest.clearAllMocks();
		mockUseSnackbarService( { service: snackbarService } );
		setUp();
	} );

	describe( 'when the snackbar is not visible', () => {
		it( 'renders the Snackbar as not visible', () => {
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when the snackbar is visible', () => {
		it( 'renders the Snackbar as info', () => {
			showSnackbar();
			expect( component.queryByText( message ) ).not.toBeNull();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when is a success message', () => {
		it( 'renders the Snackbar as success', () => {
			act( () => {
				snackbarService.showSuccess( config );
			} );
			expect( component.queryByText( message ) ).not.toBeNull();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when is an error message', () => {
		it( 'renders the Snackbar as error', () => {
			act( () => {
				snackbarService.showSuccess( config );
			} );
			expect( component.queryByText( message ) ).not.toBeNull();
			expect( component ).toMatchSnapshot();
		} );
	} );
} );
