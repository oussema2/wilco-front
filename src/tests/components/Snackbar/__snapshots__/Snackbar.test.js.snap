// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar /> when is a success message renders the Snackbar as success 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLiveRegion="polite"
    collapsable={false}
    nativeID="animatedComponent"
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "borderRadius": 8,
        "color": "#FFFFFF",
        "elevation": 6,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 8,
        "marginBottom": 50,
        "marginHorizontal": 8,
        "minHeight": 67,
        "opacity": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 6,
        "transform": Array [
          Object {
            "scale": 0.9,
          },
        ],
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "flex": 1,
              "flexWrap": "wrap",
              "marginLeft": 16,
              "marginVertical": 14,
            },
            Object {
              "color": "#ffffff",
              "marginRight": 16,
            },
          ],
        ]
      }
    >
      A message
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`<Snackbar /> when is an error message renders the Snackbar as error 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLiveRegion="polite"
    collapsable={false}
    nativeID="animatedComponent"
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00A86F",
        "borderRadius": 8,
        "color": "#FFFFFF",
        "elevation": 6,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 8,
        "marginBottom": 50,
        "marginHorizontal": 8,
        "minHeight": 67,
        "opacity": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 6,
        "transform": Array [
          Object {
            "scale": 0.9,
          },
        ],
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "flex": 1,
              "flexWrap": "wrap",
              "marginLeft": 16,
              "marginVertical": 14,
            },
            Object {
              "color": "#ffffff",
              "marginRight": 16,
            },
          ],
        ]
      }
    >
      A message
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`<Snackbar /> when the snackbar is not visible renders the Snackbar as not visible 1`] = `null`;

exports[`<Snackbar /> when the snackbar is visible renders the Snackbar as info 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLiveRegion="polite"
    collapsable={false}
    nativeID="animatedComponent"
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "borderRadius": 8,
        "color": "#FFFFFF",
        "elevation": 6,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 8,
        "marginBottom": 50,
        "marginHorizontal": 8,
        "minHeight": 67,
        "opacity": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 6,
        "transform": Array [
          Object {
            "scale": 0.9,
          },
        ],
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "flex": 1,
              "flexWrap": "wrap",
              "marginLeft": 16,
              "marginVertical": 14,
            },
            Object {
              "color": "#ffffff",
              "marginRight": 16,
            },
          ],
        ]
      }
    >
      A message
    </Text>
  </View>
</RCTSafeAreaView>
`;
